distributed/comm/tests/test_ucx.py

```diff
@@ -0,0 +1,313 @@
+from distributed.comm.registry import backends, get_backend
+from distributed.comm import ucx, parse_address
+from distributed.protocol import to_serialize
+from distributed.deploy.local import LocalCluster
+from distributed.utils_test import gen_test, loop, inc  # noqa: 401
+
+from .test_comms import check_deserialize
+
+
+HOST = ucp.get_address()
+
+# Currently having some issues with re-using ports.
+# Tests just hang. Still debugging.
+port_counter = itertools.count(13337)
+
+
+def test_parse_address():
+    result = ucx._parse_address("ucx://10.33.225.160")
+    assert result == ("ucx", "10.33.225.160")
+
+
+def test_parse_host_port():
+    assert ucx._parse_host_port("10.33.225.160:13337") == ("10.33.225.160", 13337)
+    assert ucx._parse_host_port("10.33.225.160:13338") == ("10.33.225.160", 13338)
+
+
+def test_registered():
+    assert "ucx" in backends
+    backend = get_backend("ucx")
+    assert isinstance(backend, ucx.UCXBackend)
+
+
+async def get_comm_pair(listen_addr, listen_args=None, connect_args=None, **kwargs):
+    q = asyncio.queues.Queue()
+
+    async def handle_comm(comm):
+        await q.put(comm)
+
+    # Workaround for hanging test in
+    # pytest distributed/comm/tests/test_ucx.py::test_comm_objs -vs --count=2
+    # on the second time through.
+    ucp._libs.ucp_py.reader_added = 0
+
+    listener = listen(listen_addr, handle_comm, connection_args=listen_args, **kwargs)
+    with listener:
+        comm = await connect(
+            listener.contact_address, connection_args=connect_args, **kwargs
+        )
+        serv_com = await q.get()
+        return comm, serv_com
+
+
+@pytest.mark.asyncio
+async def test_ping_pong():
+    address = "ucx://{}:{}".format(HOST, next(port_counter))
+    com, serv_com = await get_comm_pair(address)
+    msg = {"op": "ping"}
+    await com.write(msg)
+    result = await serv_com.read()
+    assert result == msg
+    result["op"] = "pong"
+
+    await serv_com.write(result)
+
+    result = await com.read()
+    assert result == {"op": "pong"}
+
+    await com.close()
+    await serv_com.close()
+
+
+@pytest.mark.asyncio
+async def test_comm_objs():
+    address = "ucx://{}:{}".format(HOST, next(port_counter))
+    comm, serv_com = await get_comm_pair(address)
+
+    assert comm.peer_address == address
+    scheme, loc = parse_address(comm.peer_address)
+    assert scheme == "ucx"
+
+    assert comm.peer_address == address
+    scheme, loc = parse_address(serv_com.peer_address)
+    assert scheme == "ucx"
+
+
+def test_ucx_specific():
+    """
+    Test concrete UCX API.
+    """
+    # TODO:
+    # 1. ensure exceptions in handle_comm fail the test
+    # 2. Use dict in read / write, put seralization there.
+    # 3. Test peer_address
+    # 4. Test cleanup
+    async def f():
+        address = "ucx://{}:{}".format(HOST, next(port_counter))
+
+        async def handle_comm(comm):
+            # XXX: failures here don't fail the build yet
```

Comment:
I think by "build" I meant "test run". It'd be good to check if this is still true by adding an `assert 0` in this `handle_comm`.

Reply:
Done